src/main/surfrdUtilsMod.F90

```diff
@@ -145,6 +146,83 @@ subroutine convert_cft_to_pft( begg, endg, cftsize, wt_cft )
+                               isturb_md
+    !
+    ! !ARGUMENTS:
+    integer, intent(in) :: begg  ! Beginning grid cell index
+    integer, intent(in) :: endg  ! Ending grid cell index
+    real(r8), intent(in) :: toosmall_soil  ! Soil landunit threshold (%)
+    real(r8), intent(in) :: toosmall_crop  ! Crop landunit threshold (%)
+    real(r8), intent(in) :: toosmall_glacier  ! Glacier landunit threshold (%)
+    real(r8), intent(in) :: toosmall_lake  ! Lake landunit threshold (%)
+    real(r8), intent(in) :: toosmall_wetland  ! Wetland landunit threshold (%)
+    real(r8), intent(in) :: toosmall_urban  ! Urban landunits threshold (%)
+    ! This array modified in-place
+    ! Weights of landunits per grid cell
+    real(r8), intent(inout) :: wt_lunit(begg:endg, max_lunit)
+    !
+    ! !LOCAL VARIABLES:
+    integer :: g  ! grid cell indexes
+    integer :: m  ! landunit indexes
+    integer :: max_landunit  ! landunit with largest fraction
+    real(r8) :: toosmall_any  ! sum of the landunit thresholds
+    real(r8) :: toosmall(max_lunit)  ! Array of the thresholds (fraction)
+    real(r8) :: residual(max_lunit)  ! Array of wt_lunit residuals (fraction)
+    !-----------------------------------------------------------------------
+
+    SHR_ASSERT_ALL((ubound(wt_lunit) == (/endg, max_lunit/)), errMsg(sourcefile, __LINE__))
+
+    toosmall_any = toosmall_soil + toosmall_crop + toosmall_glacier + &
+                   toosmall_lake + toosmall_wetland + toosmall_urban
+
+    if (toosmall_any > 0._r8) then
+
+       ! Copy the user-defined percent thresholds into array of fractions
+       toosmall(istsoil) = toosmall_soil / 100._r8
+       toosmall(istcrop) = toosmall_crop / 100._r8
+       toosmall(istice_mec) = toosmall_glacier / 100._r8
+       toosmall(istdlak) = toosmall_lake / 100._r8
+       toosmall(istwet) = toosmall_wetland / 100._r8
+       toosmall(isturb_tbd) = toosmall_urban / 100._r8
+       toosmall(isturb_hd) = toosmall_urban / 100._r8
+       toosmall(isturb_md) = toosmall_urban / 100._r8
+
+       ! Loop through gridcells and landunits
+       do g = begg, endg
+          residual = 0._r8  ! initialize
+          do m = 1, max_lunit
+             ! Remove landunits that are too small
+             if (wt_lunit(g,m) > 0._r8 .and. wt_lunit(g,m) <= toosmall(m)) then
+                residual(m) = wt_lunit(g,m)
+                wt_lunit(g,m) = 0._r8
+             end if
+          end do
+          ! If all landunits got removed, go back and keep the largest landunit
+          if (sum(wt_lunit(g,:)) == 0._r8) then
+             max_landunit = maxloc(residual, 1)
+             wt_lunit(g,max_landunit) = residual(max_landunit)
+             write(iulog,*) 'WARNING: The values of namelist parameters toosmall_* have resulted in the removal of all landunits in grid cell g = ', g, '. The model requires at least one landunit per grid cell, so the model has put back in this grid cell the largest landunit.'
```

Comment:
I think we should check with Mike here on if this should be a warning or an error. @barlage if the use of too_small_* and n_dom_* results in the removal of all land-units -- should the model print a warning and add back the largest land-unit -- or should it die? If it dies the user could then go back and adjust their values in order to get something that does work. This way they know for sure what's happening. Whereas if it's a warning, it could scroll past and they don't see it. Either way is valid, it just depends on how you want it to work.

Note, also @negin513

Reply:
Hmm, I would lean toward warning, mainly because I could see someone getting caught in an iteration loop trying to get a valid domain. Just need to make it clear in the documentation what the fallback procedure is. @slevisconsulting I don't suppose we have a count for the number of gridcells where this happens?

Reply:
OK, let's go with the warning.

It's going to print it out each time this happens, so for every gridcell, as is. We could ask Sam to do a summary print at the bottom, but unless you get really fancy it would be for each task.  So it potentially could be a ream of output (although most tasks would go into cesm.log rather than lnd.log). So is a warning for each gridcell OK, or do you want us to do the work to do a summary? I'm thinking it wouldn't be worth it. But, I do acknowledge that the output this warning will log could be a lot....

Reply:
@barlage I think Sam should go ahead with this as this is done. We still want to hear your opinion on the fact that the log output from this warning could get huge. But, I'm thinking that we should move forward with the current solution. Doing a summary of the output could be a time-consuming tangent.

Reply:
It's not worth any extra effort. I was just wondering if we could reduce output if a summary were available. I'm fine with the procedure as implemented.

Reply:
You could simplify it a bit, by doing a summary of each processor. But, you'd still have separate output from each processor, and you are likely to use hundreds to thousands or processors so I'm not convinced that would buy you a lot. So I think it's best to start using it, and have people let us know how much output we get from it. If it is shown to be a problem, then we can implement a solution that makes the most sense based on what we see.